docs/apb-filter-design.md

```diff
@@ -0,0 +1,80 @@
+# APB Filter Design Doc
+
```

Comment:
This is distinctly a "Spec" or a "Design" doc that we started doing this sprint for the major features. The intent is to have a plan that has been vetted and reviewed before implementation. It was a bit awkward because we were still figuring out how we wanted to accomplish this, but its the same type of document as the one @shawn-hurley sent out for multi-reg and @jmrodri sent out for auth. I think we're trying to settle on posting them as PRs against his repo moving forwards, the auth one being a good reference.

I do still need to document white/blacklist usage in the relevant config doc though.